rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/dateFilter.vue

```diff
@@ -30,20 +36,21 @@
     </div>
   </div>
 </template>
-<script>
+<script lang="ts">
 import { defineComponent } from "vue";
 import _ from "lodash";
 import DateTimePicker from "./dateTimePicker.vue";
+import { ModelValue } from "./tests/type";
 
 export default defineComponent({
   components: {
     DateTimePicker,
   },
   props: {
     modelValue: {
-      type: Object,
-      required: true
-    }
+      type: Object as () => ModelValue,
```

Comment:
the correct way to declare the prop is:

`type: Object as PropType<ModelValue>`

but recommend to use DateFilterProps instead of modelValue